ci: add `arm64` and `amd64` portable binaries to `winget`

matifali opened the pull request:
This PR updates the `release.yaml` workflow to automate updates for `arm64` and `x64` zip installers to winget. This has recently been merged into [winget](https://github.com/microsoft/winget-pkgs/pull/129175).

Thanks to @mdanish-kh for the upstream PR.

matifali committed:
ci: add arm64 and amd64 portable binaries to winget 

This PR updates `release.yaml` workflow to automate updates for `arm64` and `x64` zip installers to winget. This has recently been merged into [winget](https://github.com/microsoft/winget-pkgs/pull/129175).

Thanks to @mdanish-kh for the upstream PR.

matifali committed:
fixup!

deansheather commented on .github/workflows/release.yaml:
```diff
@@ -454,7 +464,9 @@ jobs:
             --submit `
             --version "${version}" `
-            --urls "${installer_url}|X64" `
+            --urls "${x64_installer_url}|X64" `
+            --urls "${x64_zip_url}|X64" `
+            --urls "${arm64_zip_url}" `
```
How does this work? You can specify 3 URLs (two with the same `|X64` and it just works? That's a bit too magic for my tastes...

matifali replied:
Yes, it works like that
https://github.com/microsoft/winget-create/blob/main/doc/update.md#usage-examples

We do not need extra `--urls` flags. Thanks for pointing.

matifali commented on .github/workflows/release.yaml:
```diff
@@ -454,7 +464,9 @@ jobs:
-            --urls "${installer_url}|X64" `
+            --urls "${x64_installer_url}|X64" `
+            --urls "${x64_zip_url}|X64" `
+            --urls "${arm64_zip_url}" `
```
We do not `arm64` as `wingetcreate` can detect `arm64` from the URL for first two x64 links it can not.

deansheather replied:
You should still specify it. We should not rely on this detection when all it takes is a few characters

matifali replied:
`wingetcreate` does not need architecture override anymore as it now supports parsing the URL for `amd64` and correctly marking it as x64 architecture. 

Reference: 
1. https://github.com/microsoft/winget-create/blob/08baf0e61e62dabcb2487397984fc69fad6a7499/src/WingetCreateCore/Common/PackageParser.cs#L594
2. PR: https://github.com/microsoft/winget-create/pull/445
2. This has been available since version https://github.com/microsoft/winget-create/releases/tag/v1.5.3.0

matifali committed:
remove extra `--urls` flags

matifali committed:
remove architecture override.

`wingetcreate` does not need architecture override as it now supports parsing the URL for `amd64` and correctly marking it as x64 architecture. 

Reference: 
1. https://github.com/microsoft/winget-create/blob/08baf0e61e62dabcb2487397984fc69fad6a7499/src/WingetCreateCore/Common/PackageParser.cs#L594C56-L594C61
2. PR: https://github.com/microsoft/winget-create/pull/445
2. This has been available since version https://github.com/microsoft/winget-create/releases/tag/v1.5.3.0

matifali committed:
fixup!

matifali committed:
Update release.yaml
